github/repos_releases.go

```diff
@@ -203,6 +203,10 @@ type repositoryReleaseRequest struct {
 //
 //meta:operation POST /repos/{owner}/{repo}/releases
 func (s *RepositoriesService) CreateRelease(ctx context.Context, owner, repo string, release *RepositoryRelease) (*RepositoryRelease, *Response, error) {
+	if release == nil {
```

Comment:
One might question why we don't simply pass `release` by value here (and elsewhere).
Certainly this is just due to habit (or copy/paste) and not for any performance-critical reasons.
Yes, changing it now would be a breaking API change.
Thoughts?

Reply:
I agree, that better is to use values here and elsewhere. But this is a large refactoring that break a lot.

For this specific case, `*RepositoryRelease`. The struct is relatively large (~15+ fields) but most fields are pointers anyway. So, performance difference would be negligible for typical API usage. And value semantics would actually be cleaner here, since the function doesn't modify the input.

Also, in the `CreateRelease` better to introduce a separate struct with only necessary for creation fields, e.g. `CreateRepositoryRelease`.

Reply:
> I agree, that better is to use values here and elsewhere. But this is a large refactoring that break a lot.

OK, do you want to continue with this PR as-is and make a separate issue/PR for breaking many APIs and switching to value semantics where the args are actually required? (I do prefer value semantics for required args, but fear that I got lazy in my code reviews over the years and allowed pass-by-reference where pass-by-value would have been a better choice.)

I'm totally fine either way.

Reply:
Created #3644